test/modules/ajax-tiles.js

```diff
@@ -49,7 +49,15 @@
                 loadTilesWithAjax: true,
                 ajaxHeaders: {
                     'X-Viewer-Header': 'ViewerHeaderValue'
-                }
+                },
+                // TODO: this test proves that tile cacheKey does not change
+                //  with headers change, which by default are part of the key,
+                //  but we cannot automatically update them since users might
+                //  manually change it long before... or can we? The tile gets
+                //  reloaded, so user code should get re-executed. IMHO,
+                //  with tile propagation the best would be throw away old tile
+                //  and start anew
+                callTileLoadedWithCachedData: true
```

Comment:
Should the headers change either update the tile keys or clear cache as well, since we don't update the cache key and thus potentially have incorrect state? I think the only state-consistent way of doing this is by unloading all the tiles this image owns, and let the system pull them anew. Whether the headers are reflected on the key later is again up to the system (overriden method for cache generation?).

Reply:
Interesting. The downside to throwing tiles away is they would have to be reloaded even if it's all the same pixels, right? It seems like it would be nice to avoid that if possible, but I'm fuzzy on what scenarios would require this.

Reply:
This might be actually outdated, will have to properly check. The comment gues a year back :D